src/main_controller.ts

```diff
@@ -104,6 +105,7 @@ export class MainController implements vscode.Disposable {
         this.client.on("request", this.onNeovimRequest);
         this.setClientInfo();
         await this.client.setVar("vscode_channel", await this.client.channelId);
+        await this.client.setVar("vscode_nvim_min_version", NVIM_MIN_VERSION);
```

Comment:
instead of adding more global vars can we change `g:vscode` to be a dict, and store things in its keys

Reply:
But it’s not convenient to update some data, right? Sometimes variables are not defined at the same time

Reply:
untested, but just calling `vim.g.vscode = vim.g.vscode or {}` first is a good enough pattern (eventually Nvim core should make this easier):

```typescript
await this.client.lua("vim.g.vscode = vim.g.vscode or {}");
await this.client.lua("vim.g.vscode.channel = ...", await this.client.channelId);
await this.client.lua("vim.g.vscode.nvim_min_version", = ...", NVIM_MIN_VERSION);
```

Reply:
In this case, there are quite a few places that need to be modified. If refactoring is required, a separate PR would be more appropriate.

Reply:
> Untested:

I just tested it and it doesn't work properly

Reply:
> I just tested it and it doesn't work as expected.

what exactly does that mean?

Reply:
![image](https://github.com/vscode-neovim/vscode-neovim/assets/47070852/3938158c-2c66-41d0-b2df-44dc7405eb68)

Reply:
Oh that's because the container is marshalled. So it will need to be vimscript intsead of lua.